core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/jdbc/UniqueIDJDBCUserStoreManager.java

```diff
@@ -5191,4 +5169,118 @@ private void deleteMultiValuedAttributes(Connection dbConnection, String userId,
+        addingWheres(header, subSqlBuilder);
+        buildClaimWhereConditions(subSqlBuilder, expressionCondition.getAttributeName(),
+                expressionCondition.getOperation(), expressionCondition.getAttributeValue());
+
+        return subSqlBuilder;
+    }
+
+    /**
+     * Merge multiple MySQL sub-queries into one by joining them on UM_USER_ID.
+     *
+     * @param mainSqlBuilder      Contains the username-filtered query when isUsernameFiltering is true.
+     * @param subSqlBuilders      Queries for role/claim filters to be joined.
+     * @param isUsernameFiltering True if username filtering is enforced.
+     * @return A single SqlBuilder combining all filters via INNER JOINs.
+     */
+    private SqlBuilder buildMySqlCombinedSqlBuilder(SqlBuilder mainSqlBuilder, List<SqlBuilder> subSqlBuilders,
+                                                    boolean isUsernameFiltering) {
+
+        SqlBuilder baseBuilder;
+        int startIdx;
+
+        // Select which query becomes the primary "t1".
+        if (isUsernameFiltering) {
+            // Use the pre-filtered main query as t1.
+            baseBuilder = mainSqlBuilder;
+            startIdx = 0;
+        } else {
+            if (subSqlBuilders.size() == 1) {
+                // If there's only one sub-query and no username filter, return it as is—no need to join.
+                return subSqlBuilders.get(0);
+            }
+            // Otherwise, the first sub-query is t1; join the rest.
+            baseBuilder = subSqlBuilders.get(0);
+            startIdx = 1;
+        }
+
+        // Initiate constructing the final SQL by wrapping baseBuilder as alias t1.
+        StringBuilder combinedSqlBuilder = new StringBuilder()
+                .append("SELECT t1.UM_USER_ID, t1.UM_USER_NAME FROM (")
+                .append(baseBuilder.getQuery())
+                .append(") AS t1");
+
+        // Collect parameters from baseBuilder in order.
+        List<Object> orderedParams = new ArrayList<>(baseBuilder.getOrderedParameters());
+
+        // Append an INNER JOIN for each remaining sub-query.
+        int alias = 2;
+        for (int i = startIdx; i < subSqlBuilders.size(); i++) {
+            SqlBuilder sub = subSqlBuilders.get(i);
+            combinedSqlBuilder.append(" INNER JOIN (")
+                    .append(sub.getQuery())
+                    .append(") AS t").append(alias)
+                    .append(" ON t1.UM_USER_ID = t").append(alias).append(".UM_USER_ID");
+
+            // Merge this sub-query’s parameters to preserve placeholder order
+            orderedParams.addAll(sub.getOrderedParameters());
+            alias++;
+        }
+
+        // Wrap into a new SqlBuilder and supply all collected parameters
+        SqlBuilder combined = new SqlBuilder(new StringBuilder(combinedSqlBuilder.toString()));
+        combined.appendParameterizedSqlFragment("", orderedParams);
+
```

Comment:
Remove the new line.

Reply:
Resolved via https://github.com/wso2/carbon-kernel/pull/4305/commits/ac37534bb69427c5149b7558b89fd10120e62ad9

Reply:
Resolved via https://github.com/wso2/carbon-kernel/pull/4305/commits/ac37534bb69427c5149b7558b89fd10120e62ad9